libs/api/libbpf_link.cpp

```diff
@@ -39,10 +40,29 @@ bpf_link__unpin(struct bpf_link* link)
 
     result = ebpf_object_unpin(link->pin_path);
     if (result != EBPF_SUCCESS) {
-        return libbpf_err(-result);
+        return libbpf_result_err(result);
     }
 
     free(link->pin_path);
     link->pin_path = nullptr;
     return 0;
 }
+
+int
+bpf_link__fd(const struct bpf_link* link)
+{
+    return link->fd;
+}
+
+int
+bpf_link_detach(int link_fd)
+{
+    struct bpf_link* link;
+    ebpf_result_t result = ebpf_get_link_by_fd(link_fd, &link);
```

Comment:
Why do we need to implement `ebpf_get_link_by_fd()`? There is already `ebpf_link_detach()` function which gets the handle from the bpf_link struct and makes the DETACH IOCTL call. We can simply add a new function `ebpf_link_detach_by_fd()` parallel to `ebpf_link_detach()` which will get the handle from the fd  and call DETACH IOTCL. We can then call the new function `ebpf_link_detach_by_fd` here.

Creating a new link object internally and caching it has problems -- since the UM app never got the link object and was only working on link fd, it will directly call `close()` on the fd later, and the cached `bpf_link` object will be leaked.


ebpf_link_detach

Reply:
Good point, done